### DISCUSSION TITLE
[Bug]: As of 2038, none default helicopter is available anymore

### DISCUSSION THREAD

**Discussion by LC-Zorg** [OpenTTD/OpenTTD#11737] [Features]
### Version of OpenTTD

12.2 - an old problem, possibly also existing in the original game

### Expected result

As with other vehicles, at least one helicopter is always available here too - the Guru X2 as one of the two helicopters is never expire. 

### Actual result

In 2038, the second Guru X2 helicopter in the basic game is withdrawn, which means that no helicopter is available anymore. If someone wants to continue the game after 2050 and uses helicopters, he will not be able to buy new ones or renew the ones he has, which in the case of turned on failures will result in their constant breakdowns and information about the need to replace them. 

Recently, someone raised this issue on the tt-forums ([here](https://www.tt-forums.net/viewtopic.php?p=1255584#p1255584)) and also, maybe the same person or not, on the Steam ([here](https://steamcommunity.com/app/1536610/discussions/0/3365901132003422373/)). It is also an issue that I wanted to report some time ago.

### Steps to reproduce

Start the game with no add-ons, with the default setting, feel as if you are a beginner. Enjoy the game. In 2060, when you just started to really develop, find out that all your helicopters have started to break down incessantly and you cannot repair or replace them. There is nothing you can do...

**Comment by nielsmh** [comment]
Yes that's traditional game data.

The risk with changing it is that it might cause NewGRF vehicles to change behaviour, if they inherit from the changed vehicle, and don't change the property but assume it'll have its traditional value.

**Comment by LC-Zorg** [comment]
I don't know of any NewGRF that uses the parameters of existing helicopters. The closest to that seems to be Use-able Defaulf Aircrafts, which, however, have changed the availability dates of both helicopters so that they never disappear. However, if there is an add-on, it must be very niche and most likely unavailable in the online content. And even if I'm wrong and such an add-on is available, such a change will not spoil anything. Because what could spoil the extension of the availability of one helicopter? I do not know. Meanwhile, their unexpected withdrawal certainly surprised many players and spoiled (and will spoil) many games.

**Comment by LC-Zorg** [comment]
> Yes that's traditional game data.

Ok, but note that traditionally the game ended in 2050. Back then, those few years when you couldn't buy or repair helicopters wasn't a major problem. Although looking at other types of vehicles, where one is always available everywhere, this seems more like an oversight than a deliberate design. However, today, when you can play practically endlessly, retiring one of the means of transport entirely and forcing the player to remove all his helicopters and landing pads or watch them break down non-stop for the next 300 years or more, I believe is simply a mistake of this games.

> The risk with changing it is that it might cause NewGRF vehicles to change behaviour, if they inherit from the changed vehicle, and don't change the property but assume it'll have its traditional value.

I decided to take a closer look and nowhere I didn't notice any of the helicopters in any set that using default values of withdrawal date (ca. 2041).

List of proven sets:
- av8 - two of the eight helicopters / airships are being retired before 2020, the rest remain available forever
- av8 Minimal Supp WAS - same as av8
- av9.8 - two helicopters are available, one is retired in 1994, the other never
- Hot Air Balloon - never retired
- Kaaskroket's Fictional Aircrafts - planes are never retired regardless of the setting
- OpenGFX+ Mars Aircrafts - old, unfinished set (shame); here the last helicopter is retired in 2045; probably if someone had finished it today the dates would have been completely different and just as unrelated to the defaults
- Planetset - old, public unavailable add-on; includes six helicopters/airships; retirement dates are unrelated to default values
- RAV8 - same as av8; no relation to default values
- Russian Planetet - old, public unavailable add-on; includes five helicopters; retirement dates are unrelated to default values
- Use-able Default Aircrafts - aircraft are never retired regardless of the setting
- VACE-VTOL - add-on is a supplement to the default aircraft; it does not change the dates of the default helicopters, nor does it use them
- WAS - the only airship is retired in 2048; otherwise, this is a significant disadvantage of this set, because it does not provide for a longer game

I didn't report this problem only because it was something particularly bothersome for me. This issue has been touched at least twice on the Steam forums and probably many, many times on tt-forums over the last 20 years.

It really is unpleasant when after many hours of play you find that all your work has been spoiled.

**Comment by andythenorth** [comment]
Default policy is to fix it in content.  That's not a 100% rule for every case, but the default fix here should be for $someone to make a grf and put it on the content service.

**Comment by LC-Zorg** [comment]
> Default policy is to fix it in content.

But this problem is with the game without add-ons. Why does the base game have to be broken? Note that today you can play much longer than before. This is the only group of vehicles that has never been adapted to this change.
